Assets/Scripts/Controllers/AI/Boss/Boss.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class Boss : MonoBehaviour {
+	public BossStatsSO Stats;
+	[SerializeField] public BossAttacks BossAttacks;
+	public Transform Player;
```

Comment:
Same goes for the player. Run a check to find the PlayerStateMachine or smth in the scene. This helps when transfering the boss between scenes as a prefab

Reply:
Also if there already is a reference to the player, why is the CheckForPlayer function neccesary?